datadog_checks_base/datadog_checks/base/utils/time.py

```diff
@@ -6,61 +6,41 @@
 from datetime import datetime
 from time import time as epoch_offset
 
-import pytz
-from six import PY2
+from dateutil.tz import UTC
 
-UTC = pytz.utc
-EPOCH = datetime(1970, 1, 1, tzinfo=UTC)
+EPOCH = datetime.fromtimestamp(0, UTC)
 
 
-if PY2:
-
-    def get_timestamp(dt=None):
-        """
-        Returns the number of seconds since the Unix epoch.
-        If `dt` is not specified or `None`, the current time in UTC is assumed.
-        """
-        if dt is None:
-            return epoch_offset()
-
-        return (dt - EPOCH).total_seconds()
-
-
-else:
-
-    def get_timestamp(dt=None):
-        """
-        Returns the number of seconds since the Unix epoch.
-        If `dt` is not specified or `None`, the current time in UTC is assumed.
-        """
-        if dt is None:
-            # NOTE: Although the epoch is platform dependent, it appears to be the same
-            # for all platforms we've tested, therefore we use `time.time` for speed.
-            #
-            # Here is the test:
-            #
-            #     $ python -c "import time;print(tuple(time.gmtime(0)[:9]))"
-            #     (1970, 1, 1, 0, 0, 0, 3, 1, 0)
-            #
-            # If you can reproduce, add to the following list of tested platforms:
-            #
-            # - Windows
-            # - macOS
-            # - Ubuntu
-            # - Alpine
-            return epoch_offset()
-
-        return normalize_datetime(dt).timestamp()
-
-
-def get_aware_datetime(dt=None, tz=UTC):
+def get_timestamp(dt=None):
     """
-    Returns an aware datetime object. If `dt` is not specified or `None`, the current time in UTC is assumed.
+    Returns the number of seconds since the Unix epoch.
+    If `dt` is not specified or `None`, the current time in UTC is assumed.
     """
     if dt is None:
-        return datetime.now(tz)
-    else:
-        return normalize_datetime(dt)
+        # NOTE: Although the epoch is platform dependent, it appears to be the same
+        # for all platforms we've tested, therefore we use `time.time` for speed.
+        #
+        # Here is the test:
+        #
+        #     $ python -c "import time;print(tuple(time.gmtime(0)[:9]))"
+        #     (1970, 1, 1, 0, 0, 0, 3, 1, 0)
+        #
+        # If you can reproduce, add to the following list of tested platforms:
+        #
+        # - Windows
+        # - macOS
+        # - Ubuntu
+        # - Alpine
+        return epoch_offset()
+
+    return (normalize_datetime(dt) - EPOCH).total_seconds()
```

Comment:
I think using `.timestamp()` is almost certainly better on Python 3, but I suppose I can see why you'd want to unify the implementation if you're still supporting Python 2.

Reply:
Yes I agree the official method is preferred 🙂 We'll definitely switch when we drop Python 2

Reply:
We should add a comment to identify this as a possible optimization for when we drop Python 2.

Reply:
Added!

Reply:
Probably nitpicking, but alternatively, to benefit from the standard behavior on Python 3, use the old strategy on Python 2, and make it clear that we can drop the Py2 branch in due time without a TODO that might go out of date, we could use something like this in a `compat.py` module:

```python
# datadog_checks_base/datadog_checks/base/utils/compat.py

if PY2:

    def to_timestamp(dt):
        return (normalize_datetime(dt) - EPOCH).total_seconds()

else:

    def to_timestamp(dt):
        return normalize_datetime(dt).timestamp()
```

(We might benefit from introducing such a `compat.py` like this anyway, eg for imports in `contextlib` vs `contextlib2`, etc. But I agree that'd be a precedent to make, so not too fussed about the current impl.)

Reply:
I had it like that https://github.com/DataDog/integrations-core/pull/6663#discussion_r426449365

@AlexandreYang ?

Reply:
In our case, `dt.timestamp()` in py3 is just a shortcut for `(self - _EPOCH).total_seconds()`.

https://github.com/python/cpython/blob/3.8/Lib/datetime.py#L1794

I guess both are fine for me, up to you @ofek :)

Reply:
> When the time comes

Hopefully not too late ;)